adapters/aidem/aidem.go

```diff
@@ -137,6 +141,21 @@ func getImpressionExt(imp *openrtb2.Imp) (*openrtb_ext.ExtImpAidem, error) {
 	return &AIDEMExt, nil
 }
 
+func getSupportedImps(imps []openrtb2.Imp) ([]openrtb2.Imp, error) {
+	var supportedImps []openrtb2.Imp
+	for _, imp := range imps {
+		if imp.Banner != nil || imp.Video != nil {
+			supportedImps = append(supportedImps, imp)
+		}
+	}
+	if len(supportedImps) == 0 {
+		return nil, &errortypes.BadInput{
+			Message: "No supported formats (banner or video) found",
+		}
+	}
+	return supportedImps, nil
+}
+
```

Comment:
No need to check this in the adapter code. You already mentioned supported types in the `static/bidder-info/aidem.yaml` and PBS core will perform this validation. Adapter code will only receive banner and video imps.